### ISSUE TITLE
Track changes and progress of issues before we tag 0.2.0

### ISSUE THREAD

**Issue by gits7r** [torproject/onionbalance#25]
This is a master ticket just to keep track of what was already done and what is left to do before we tag 0.2.0.

- [x] Support Tor HSv3 private key file format: Now you can migrate your existing v3 onion service to onionbalance. Fixes #10 --- **this still needs #24 to be fixed**;

- [ ] Do not crash while running if at least 1 (one) backend instance is still running fine. This should protect the frontend from some attacks. Fixes #20

- [x] Do not crash while running if communication with Tor control port stops suddenly or becomes too slow to use, wait for it to come back. Fixes #15

- [x] The configuration file can now be changed on the fly and reloaded by sending a SIGHUP to the onionbalance process. Fixes #19

- [x] Improve log messages and add some more diagnostics for descriptor uploads results. Fixes #16

- [x] Other documentation improvementst and small fixes for onionbalance-config interactive configuration file generator. Fixes #22, #24

- [x] Minor additional checks for the config file before we start up. Fixes #26, #27 

- [ ] Status socket support for v3 onion services. Fixes #18 --- **if we can cherry pick something here OK, otherwise we can leave #18 for the next release, not a blocker***

**Comment by asn-d6** [comment]
Hello I have addressed various parts of the above checkpoints. I don't think we need all of them to tag 0.2.0 (e.g. we are not gonna get status socket support yet), but if the things I pushed actually work, I think it would be nice to tag 0.2.0! :)

Let me know how you like them and thanks for tracking these changes!

**Comment by gits7r** [comment]
I agree 0.2.0 is good to tag, so package maintainers can build packages since the tool is reliable and pretty stable at this moment (I am running it on multiple vms and doing stress testing + abnormal usage / simulation of misconfigurations, etc. and it only crashed two times, because of #20 which needs more information, we are not sure what it is. #18 can wait as you say).

I have added a comment on #27 for a small misbehavior, besides this I have tested everything and works as expected. While you are it, might fix in the same commit this small typo: https://github.com/asn-d6/onionbalance/commit/3ff3aa28c00cf93046d142d0f3e46234c5fa52ac#comments

Changes file needed, so we can close this ticket and tag. Nice work, super thanks for incredibly fast pushing fixes for everything!
